app/src/main/java/com/github/onlynotesswent/ui/overview/EditMarkdownScreen.kt

```diff
@@ -0,0 +1,303 @@
+    }
+  }
+
+  Scaffold(
+      topBar = {
+        ScreenTopBar(
+            titleTestTag = "modifyMDTitle",
+            onBackClick = { navigationActions.goBack() },
+            title = "Modify Markdown",
+            icon = {
+              Icon(imageVector = Icons.AutoMirrored.Outlined.ArrowBack, contentDescription = "Back")
+            },
+            iconTestTag = "goBackButton")
+      },
+      content = { paddingValues ->
+        Column(
+            modifier =
+                Modifier.fillMaxSize()
+                    .padding(16.dp)
+                    .padding(paddingValues)
+                    .testTag("editMarkdownColumn")
+                    .verticalScroll(rememberScrollState()),
+        ) {
+          EditorControls(
+              modifier = Modifier.testTag("EditorControl"),
+              state = state,
+              onBoldClick = { state.toggleSpanStyle(SpanStyle(fontWeight = FontWeight.Bold)) },
+              onItalicClick = { state.toggleSpanStyle(SpanStyle(fontStyle = FontStyle.Italic)) },
+              onUnderlineClick = {
+                state.toggleSpanStyle(SpanStyle(textDecoration = TextDecoration.Underline))
+              },
+              onStrikethroughClick = {
+                state.toggleSpanStyle(SpanStyle(textDecoration = TextDecoration.LineThrough))
+              })
+
+          RichTextEditor(
+              modifier = Modifier.fillMaxWidth().weight(2f).testTag("RichTextEditor"),
+              state = state,
+          )
+
+          Button(
+              modifier = Modifier.fillMaxWidth().testTag("Save button"),
+              onClick = { updateMarkdownFile(context, note?.id ?: "", fileViewModel) }) {
+                Text("Save")
+              }
+        }
+      })
+}
+/**
+ * A composable component that provides a set of text formatting controls for the rich text editor.
+ *
+ * @param modifier Modifier to be applied to the Row layout containing the controls.
+ * @param state The current state of the rich text editor, used to apply styles.
+ * @param onBoldClick Callback function to handle bold style toggle.
+ * @param onItalicClick Callback function to handle italic style toggle.
+ * @param onUnderlineClick Callback function to handle underline style toggle.
+ * @param onStrikethroughClick Callback function to handle strikethrough style toggle.
+ */
+@Composable
+fun EditorControls(
+    modifier: Modifier,
+    state: RichTextState,
+    onBoldClick: () -> Unit,
+    onItalicClick: () -> Unit,
+    onUnderlineClick: () -> Unit,
+    onStrikethroughClick: () -> Unit
+) {
+  var boldSelected by rememberSaveable { mutableStateOf(false) }
+  var italicSelected by rememberSaveable { mutableStateOf(false) }
+  var underlineSelected by rememberSaveable { mutableStateOf(false) }
+  var strikethroughSelected by rememberSaveable { mutableStateOf(false) }
+  var previousMarkdown by rememberSaveable { mutableStateOf("") }
+  LaunchedEffect(Unit) {
+    while (true) {
```

Comment:
Did you consider using a snapshotFlow to observe changes in the state instead of using the infinite loop to check and update the state?